### ISSUE TITLE
Suggestions for Control Center

### ISSUE THREAD

**Issue by JoshStrobl** [budgie-desktop/budgie-rd#8]
So a conversation from IRC has inspired me to jot down at least some items that, in my opinion, should be addressed in Budgie 11's Control Center. Some of these items are from that conversation, some are from my own personal experience, some are simply shortcomings in other control centers. Most of these are niceties, shouldn't ever be considered blockers, but it wouldn't hurt to write them down.

### Background

- **Animated:** This is more a feature than a shortcoming in other Control Centers (and respective display managers that may not support it), but having basic support for short looping video would be a nicety. Here's [an example](https://i.kinja-img.com/gawker-media/image/upload/s--SjiovDbQ--/c_fit,fl_progressive,q_80,w_636/fzmhvmqeoxpkwfm8hnqy.gif) of what I'm talking about (namely the beach example, nothing too crazy).

- **Color:** Background selection should have the ability to utilize a color picker under a "Colors" section to pick your own color, if any of the pre-defined options aren't suitable. You shouldn't have to modify the code of the Control Center just to have a new color to choose from.

- **Slideshow:** I imagine this would be likely more trivial to implement over "animated backgrounds", but being able to specify a set of photos (*whether that's a specific folder or some file chooser*) and time interval would be a nicety.

### Display

- **Night Mode:** The capability to set minimum and maximum temperatures as well as custom time schedule to enable night mode / blue shifting is a must really. We're no normal mortals around here, clearly when the sun is shining through my window it's time for me to go to bed and my display should know that and not burn as bright as the actual sun :smile:

### Keyboard

Some of this is part "expand on the customization of Budgie" and part "why is this a pain to do in 2017 still".

- **Layout:** There's like 12 people out there using Colemak and Dvorak layouts, but they shouldn't have to jump through hoops (or sub lists of sub lists) to get their favorite non-QWERTY keyboard layout working.
- **Remapping:** In my opinion, being able to remap basic function / modifier keys would go a long way. Examples below.
  - In GNOME Tweak Tool, you're able to swap the Caps Lock key for an modifier key, such as `Control`, `Super`, `Esc`, etc. and frankly I think this is something that should just be supported OOTB in a Control Center. Whether it's as simple as "press this input box, press the key, then click the done button" as a way of remapping or a pre-defined set of options, let's turn Caps Lock into something that's actually useful.
 - You should be able to swap between left and right Super keys for Budgie Menu, or even leveraging the Caps Lock key modifier to use that as the key. There is legitimately plenty of keyboards that have no Super keys, typically in the "minivan" form factor or custom-made (as in PCBs).

### Touchpad

- A toggle switch to disable the touchpad or ignore its input while typing would be a nicety.

**Comment by ckrzen** [comment]
This reminds me:

1. Please add QT gui option to expose/control number-of-workspaces(ala:
org.gnome.desktop.wm.preferences num-workspaces)

2. Option to change Desktop wallpaper independent of Screensaver/Lock
Screen?

3. Make the "Print Screen" button work outta-the-box or a least be
re-mappable(can't even re-map it now in Gnome 3.26 aka Ubuntu Budgie 17.10)

Thanks!

**Comment by cunidev** [comment]
My (very personal) opinion:

Those are great ideas, but don't animated backgrounds use relatively much cpu for - essentially - nothing special? Having it as a feature would be great, but it might make Budgie way heavier than it currently is, which is something to consider.
Color and Slideshow both look great, though I can't think of good enough algorithms to compute a full palette from a single color the solution might not be very difficult.
Key remapping is a feature I've always loved, would it be possible also to keep "external" Budgie/Solus app shortcuts in centralized configuration files instead of hardcoding them? Also having a fixed (default) key for "Back" actions (my candidate is Esc) in apps like solus-sc would come in handy when it comes to usability (considering I tend to use the keyboard way more than the mouse, especially on Linux)

Tl;dr: love the concept!

**Comment by Vistaus** [comment]
Great ideas!

As for the "Disable touchpad" feature: it would be even better if there was a Budgie applet for quick enable/disable or if we could just set a keyboard shortcut to toggle.
